akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
+      val probe = TestProbe[Event]()
+      val child = Behaviors.immutablePartial[Command] {
+        case (_, Stop) ⇒
+          Behaviors.stopped
+      }.decorate
+      val actor = spawn(
+        (Behaviors.immutablePartial[Command] {
+          case (ctx, MakeChild) ⇒
+            val childRef = ctx.spawn(child, "A")
+            probe.ref ! ChildMade(childRef)
+            Behaviors.same
+          case (ctx, Watch(ref)) ⇒
+            ctx.watch(ref)
+            probe.ref ! Pong
+            Behaviors.same
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
+      )
+      actor ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      childRef ! Stop
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
     }
 
     "unwatch a child actor before its termination" in {
-      sync(setup("ctx12") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith.mkChild(None, ctx.spawnMessageAdapter(ChildEvent), self).keep {
-          case (subj, child) ⇒
-            subj ! Watch(child, self)
-        }.expectMessageKeep(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(Watched)
-            subj ! Unwatch(child, self)
-        }.expectMessage(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(Unwatched)
-            ctx.watch(child)
-            child ! Stop
-            child
-        }.expectTermination(expectTimeout) { (t, child) ⇒
-          t should ===(Terminated(child)(null))
-        }
-      })
+      val probe = TestProbe[Event]()
+      val child = Behaviors.immutablePartial[Command] {
+        case (_, Stop) ⇒
+          Behaviors.stopped
+      }.decorate
+      val actor = spawn(
+        (Behaviors.immutablePartial[Command] {
+          case (ctx, MakeChild) ⇒
+            val childRef = ctx.spawn(child, "A")
+            probe.ref ! ChildMade(childRef)
+            Behaviors.same
+          case (ctx, Watch(ref)) ⇒
+            ctx.watch(ref)
+            probe.ref ! Pong
+            Behaviors.same
+          case (ctx, UnWatch(ref)) ⇒
+            ctx.unwatch(ref)
+            probe.ref ! Pong
+            Behaviors.same
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
+      )
+      actor ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      actor ! UnWatch(childRef)
+      probe.expectMessage(Pong)
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      childRef ! Stop
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
```

Comment:
This does not look like the original test, which watched, unwatched, and then stopped the child. It should not watch again after unwatching, but instead verify that there is no `Terminated` message coming (since the actor was unwatched)

Reply:
Actually in the original test before stopping the child they do `ctx.watch(child)` and after that `t should ===(Terminated(child)(null))`. However I agree that it does not quite match with the name of the test therefore will follow your suggestion.